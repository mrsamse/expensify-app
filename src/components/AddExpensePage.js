import React from "react";

const AddExpensePage = () => <div>This ist a 2</div>;

export default AddExpensePage;
