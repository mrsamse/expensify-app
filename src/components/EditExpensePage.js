import React from "react";

const EditExpensePage = () => <div>This ist a 3</div>;

export default EditExpensePage;
