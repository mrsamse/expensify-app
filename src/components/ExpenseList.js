import React from "react";
import ExpenseListItem from "./ExpenseListItem";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";

const ExpenseList = ({ expenses }) => (
  <div>
    <h1>ExpenseList</h1>
    {expenses.map(expense => (
      <ExpenseListItem key={expense.id} {...expense} />
    ))}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
