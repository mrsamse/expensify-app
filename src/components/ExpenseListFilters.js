import React from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate
} from "../actions/filters";
import { DateRangePicker } from "react-dates";

class ExpenseListFilters extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      focusedInput: null,
      error: ""
    };
  }

  onDatesChange = ({ startDate, endDate }) => {
    console.log("asd");
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  onFocusChange = focusedInput => {
    this.setState(() => ({ focusedInput }));
  };

  render = () => (
    <div>
      <input
        type="text"
        value={this.props.filters.text}
        onChange={e => {
          this.props.dispatch(setTextFilter(e.target.value));
        }}
      />
      <select
        value={this.props.filters.sortBy}
        onChange={e => {
          e.target.value === "date"
            ? this.props.dispatch(sortByDate())
            : this.props.dispatch(sortByAmount());
        }}
      >
        <option value="date">Date</option>
        <option value="amount">Amount</option>
      </select>

      <DateRangePicker
        startDate={this.props.filters.startDate}
        startDateId="startdate"
        endDate={this.props.filters.endDate}
        endDateId="enddate"
        onDatesChange={this.onDatesChange}
        focusedInput={this.state.focusedInput}
        onFocusChange={this.onFocusChange}
        numberOfMonths={1}
        isOutsideRange={() => false}
        showClearDates={true}
      />
    </div>
  );
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(ExpenseListFilters);
