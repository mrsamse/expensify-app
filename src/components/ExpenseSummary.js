import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totaling{" "}
        {numeral(expensesTotal / 100).format("$0,0.00")}
      </h1>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
