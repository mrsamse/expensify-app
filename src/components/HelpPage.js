import React from "react";

const HelpPage = () => <div>This ist a 4</div>;

export default HelpPage;
