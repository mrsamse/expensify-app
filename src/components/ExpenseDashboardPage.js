import React from "react";
import ExpenseSummary from "./ExpenseSummary";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpenseDashboardPage = props => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
