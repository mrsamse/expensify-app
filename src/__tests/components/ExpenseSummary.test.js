import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";

test("should render ExpenseSummary correctly with 1 expense", () => {
  const wrapper = shallow(
    <ExpenseSummary expenseCount={1} expensesTotal={250} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary correctly with multiple expenses", () => {
  const wrapper = shallow(
    <ExpenseSummary expenseCount={23} expensesTotal={2500} />
  );
  expect(wrapper).toMatchSnapshot();
});
