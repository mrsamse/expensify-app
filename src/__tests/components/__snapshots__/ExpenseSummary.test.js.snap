// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling
     
    $2.50
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totaling
     
    $25.00
  </h1>
</div>
`;
